stream-video-android-core/src/main/kotlin/io/getstream/video/android/core/notifications/internal/service/CallService.kt

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2014-2023 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-video-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.video.android.core.notifications.internal.service
+
+import android.app.Notification
+import android.app.Service
+import android.content.Context
+import android.content.Intent
+import android.content.pm.ServiceInfo
+import android.os.Build
+import android.os.IBinder
+import androidx.core.app.NotificationManagerCompat
+import androidx.core.app.ServiceCompat
+import io.getstream.log.taggedLogger
+import io.getstream.video.android.core.RingingState
+import io.getstream.video.android.core.StreamVideo
+import io.getstream.video.android.core.notifications.NotificationHandler
+import io.getstream.video.android.core.notifications.NotificationHandler.Companion.INTENT_EXTRA_CALL_CID
+import io.getstream.video.android.core.notifications.NotificationHandler.Companion.INTENT_EXTRA_CALL_DISPLAY_NAME
+import io.getstream.video.android.model.StreamCallId
+import io.getstream.video.android.model.streamCallDisplayName
+import io.getstream.video.android.model.streamCallId
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+import org.openapitools.client.models.CallEndedEvent
+import org.openapitools.client.models.CallRejectedEvent
+import java.lang.IllegalArgumentException
+
+/**
+ * A foreground service that is running when there is an active call.
+ */
+internal class CallService : Service() {
+    private val logger by taggedLogger("CallService")
+    private var callId: StreamCallId? = null
+    private var callDisplayName: String? = null
+    private var observeRingingState: Job? = null
+    private var observeCallState: Job? = null
+    private var updateCallJob: Job? = null
+    private var startCoordinatorSocketJob: Job? = null
+    private val supervisorJob = SupervisorJob()
```

Comment:
Hey @aleksandar-apostolov,

WDYT about creating a `serviceScope` here and use it instead of a `GlobalScope` later on? 🙂 
```
val serviceScope = CoroutineScope(SupervisorJob() + Dispatchers.IO)
```

Reply:
Its a great idea. Tnx for pointing it out.
We do not even need the SupervisorJob in this case.